### PR TITLE
Ensure nodeSelector logic is consistent for all operators

### PR DESCRIPTION
**Pull request by olliewalsh**
nodeSelectors are not currently applied consistently across all operators.

Some do not support nodeSelectors at all - will be implemented in this series of pull requests.
Some do not apply them to every pod (jobs/cronjobs esp) - will be resolved in this series of pull requests.
Some do not apply a node selector update correctly, only when not initially set.
Some support nodeSelectors but do not inherit the default nodeSelector from the OpenstackControlPlane CR.

Jira: [OSPRH-10734](https://issues.redhat.com//browse/OSPRH-10734)

### CODE REVIEW

File: test/functional/base_test.go [hunk 1]
```diff
@@ -230,6 +231,14 @@ func GetCinderVolume(name types.NamespacedName) *cinderv1.CinderVolume {
 	return instance
 }
 
+func GetCronJob(name types.NamespacedName) *batchv1.CronJob {
```
Comment (stuggi) [inline test/functional/base_test.go:234]: in a follow up should add this to https://github.com/openstack-k8s-operators/lib-common/tree/main/modules/common/test/functional

Comment (olliewalsh) [reply test/functional/base_test.go:234]: yes, also for keystone where this came from

**Review by stuggi — APPROVED** [review]
/lgtm

**Pull request merged**